### PR TITLE
🔫 Rework DB triggers to use FlowRun.status

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/api/v2/serializers.py [hunk 1]
```diff
@@ -860,9 +860,12 @@ class Meta:
...
     EXIT_TYPES = {
-        FlowRun.EXIT_TYPE_COMPLETED: "completed",
-        FlowRun.EXIT_TYPE_INTERRUPTED: "interrupted",
-        FlowRun.EXIT_TYPE_EXPIRED: "expired",
+        FlowRun.STATUS_ACTIVE: None,
+        FlowRun.STATUS_WAITING: None,
+        FlowRun.STATUS_COMPLETED: "completed",
+        FlowRun.STATUS_INTERRUPTED: "interrupted",
+        FlowRun.STATUS_EXPIRED: "expired",
+        FlowRun.STATUS_FAILED: "interrupted",  # for backwards compatibility
```
Comment (rowanseymour) [inline temba/api/v2/serializers.py]: not sure about this... can we just start returning "failed" here...?

File: temba/sql/current_functions.sql [hunk 2]
```diff
@@ -372,28 +372,43 @@ CREATE OR REPLACE FUNCTION temba_insert_flowpathrecentrun(_from_uuid UUID, _from
...
-    INSERT INTO flows_flowstartcount("start_id", "count", "is_squashed")
-    VALUES(_start_id, _count, FALSE);
-  END IF;
+  CASE _status
+  WHEN 'C' THEN
+    exit_type = 'C';
+  WHEN 'I' THEN
+    exit_type = 'I';
+  WHEN 'X' THEN
+    exit_type = 'E';
```
Comment (rowanseymour) [inline temba/sql/current_functions.sql]: leaving this as `E` for now in this PR in the name of incrementalism

**Comment by nicpottier** [comment]
Can you give us some context on what's driving this?

**Comment by rowanseymour** [comment]
Closing in favor of https://github.com/nyaruka/rapidpro/pull/3642

**Pull request closed without merging**